### PR TITLE
refactor: remove `com.google.*` dependent code from the `jkube-kit-build-api` module

### PR DESCRIPTION
**Pull request by manusa**
## Description

The only remaining usages are in the `PortMapping` class that should be addressed as part of #2481 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2484 (_2023-11-29T11:26:33Z_)

:gear: [JKube E2E Tests (7031877546)](https://github.com/jkubeio/ci/actions/runs/7031877546) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2484)](https://github.com/jkubeio/ci/actions/runs/7031877546/job/19134407366)

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 1]
```diff
@@ -446,12 +456,33 @@ public static boolean isWindows() {
         return System.getProperty("os.name").toLowerCase().contains("windows");
     }
 
+    /**
+     * Return a {@link File} pointing to the user's home directory.
+     * @return the user home directory.
+     */
+    public static File getUserHome() {
```
Comment (rohanKanojia) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:463]: Is it possible to replace this with https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SystemProperties.html#getUserHome-- ?

Comment (manusa) [reply jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:463]: This method is more complete.

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/AuthConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.Base64;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+class AuthConfigTest {
+
+  @Test
+  void simpleConstructor() {
```
[the remark this answers is not in the document]
Comment (manusa) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/AuthConfigTest.java:30]: this test is a copy-paste of RegistryAuthTest which has the exact same behavior of AuthConfigTest.

I'm unsure of the complete behavior of these classes. I simply added this test to ensure that the underlying logic behaved in the same way before and after removing the Google specific classes.

File: jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/AuthConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    AuthConfig config = AuthConfig.builder()
+      .username("roland")
+      .password("#>secrets??")
+      .email("roland@jolokia.org")
+      .build();
+    // Since Base64.decodeBase64 handles URL-safe encoding, must explicitly check
+    // the correct characters are used
+    assertThat(config.toHeaderValue(new KitLogger.SilentLogger()))
+      .isEqualTo("eyJ1c2VybmFtZSI6InJvbGFuZCIsInBhc3N3b3JkIjoiIz5zZWNyZXRzPz8iLCJlbWFpbCI6InJvbGFuZEBqb2xva2lhLm9yZyJ9");
```
[the remark this answers is not in the document]
Comment (manusa) [reply jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/auth/AuthConfigTest.java:39]: Same as https://github.com/eclipse/jkube/pull/2484#discussion_r1410823553

**Pull request merged**